pkg/lwip/contrib/sock/ip/lwip_sock_ip.c

```diff
@@ -183,55 +195,129 @@ ssize_t sock_ip_recv_aux(sock_ip_t *sock, void *data, size_t max_len,
     ssize_t res, ret = 0;
     bool nobufs = false;
 
+    bool peek = false;
+    if (aux != NULL) {
```

Comment:
```suggestion
    if ((aux != NULL) && (IS_USED(MODULE_SOCK_PEEK)) {
```

Most applications don't need to look at the same data more than once, so IMO it doesn't make sense to have everybody oay the costs for what only few applications need.

Reply:
If I understand you correctly you would like to have the whole peek functionality as a separate pseudomodule? I don't think the costs are big, but it also does not hurt to let the code remove when not needed, so that sounds reasonable to me.

Reply:
Regarding 1: Somewhere here

https://github.com/RIOT-OS/RIOT/blob/f22ba30a312ef1f835327b424f21bb2b7e446c7f/sys/net/gnrc/Makefile.dep#L78-L83

In KConfig it is a bit easier to model. E.g. it could depend on some HAS_SOCK_PEEK that is only provided by lwip.

Regarding 2: `IS_USED()` is defined in `kernel_defines.h`, you'll likely just missing an `#include` there. (I don't know why the message is so cryptic...)

Reply:
> you'll likely just missing an #include there

Thanks, it was indeed just a missing include in the build path of `sock_tcp`.

I added Makefile errors to `pkg/openwsn/Makefile.dep` and `sys/net/gnrc/Makefile.dep` when trying to use `sock_peek`.